src/common/perf_counters.h

```diff
@@ -235,6 +235,7 @@ class PerfCounters
 
   void inc(int idx, uint64_t v = 1);
   void dec(int idx, uint64_t v = 1);
+  void dec_if_positive(int idx, uint64_t v = 1);
```

Comment:
I don't think suppressing the decrement is the right approach. Is the corresponding increment skipped as well? If not while the problem is about doing the ops in wrong order (`dec()` first, then `inc()`), we would run into misaccounting.

However, I'm worried this is just a symptom of more severe issue. Could you please elaborate on how the `unregister_conn` have been reached? I'm particularly interested in the state of connections' locks (`lock`, `write_lock`).

Also, maybe the increment should happen earlier, perhaps at `AsyncMessenger::add_accept`.

Reply:
Thanks Radek for reviewing the patch. I agree with you this is not a best approach skip the decrement operation in case if it is negative. 
 For specific to unregister_conn is calling, in my reproducer as i mentioned in tracker bug it was calling in two situations.
 1) While waiting for client banner message response(block on accept_conns and then continue), the state is like below
START_ACCEPT==>SEND_SERVER_BANNER==>HANDLE_SERVER_BANNER_WRITE==>WAIT_CLIENT_BANNER==>HANDLE_CLIENT_BANNER==>FAULT
 2) In case if a server is not able to read any message for that particular socket because
    socket is already closed by client then unregister_con is call I agree add_accept is the best place to increase the counter.